src/PrivateDns/PrivateDns/Models/PSPrivateDnsRecordSet.cs

```diff
@@ -0,0 +1,583 @@
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+namespace Microsoft.Azure.Commands.PrivateDns.Models
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text;
+    using Microsoft.Azure.Management.PrivateDns.Models;
+
+    /// <summary>
+    /// Represents a set of records with the same name, with the same type and in the same zone.
+    /// </summary>
+    public class PSPrivateDnsRecordSet : ICloneable
+    {
+        /// <summary>
+        /// Gets or sets the ID of the record set.
+        /// </summary>
+        public string Id { get; set; }
+
+        /// <summary>
+        /// Gets or sets the name of this record set, relative to the name of the zone to which it belongs and WITHOUT a terminating '.' (dot) character.
+        /// </summary>
+        public string Name { get; set; }
+
+        /// <summary>
+        /// Gets or sets the name of the zone to which this recordset belongs.
+        /// </summary>
+        public string ZoneName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the name of the resource group to which this record set belongs.
+        /// </summary>
+        public string ResourceGroupName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the TTL of all the records in this record set.
+        /// </summary>
+        public uint Ttl { get; set; }
+
+        /// <summary>
+        /// Gets or sets the Etag of this record set.
+        /// </summary>
+        public string Etag { get; set; }
+
+        /// <summary>
+        /// Gets or sets the type of DNS records in this record set. Only records of this type may be added to this record set.
+        /// </summary>
+        public RecordType RecordType { get; set; }
+
+        /// <summary>
+        /// Gets or sets the list of records in this record set.
+        /// </summary>
+        public List<PSPrivateDnsRecordBase> Records { get; set; }
+
+        /// <summary>
+        /// Gets or sets the tags of this record set.
+        /// </summary>
+        public Hashtable Metadata { get; set; }
+
+        /// <summary>
+        /// Gets or sets the provisioning state of the record set
+        /// </summary>
+        public bool? IsAutoRegistered { get; set; }
+
+        /// <summary>
+        /// Returns a deep copy of this record set
+        /// </summary>
+        /// <returns></returns>
+        public object Clone()
+        {
+            var clone = new PSPrivateDnsRecordSet
+            {
+                Name = this.Name,
+                Id = this.Id,
+                ZoneName = this.ZoneName,
+                ResourceGroupName = this.ResourceGroupName,
+                Ttl = this.Ttl,
+                Etag = this.Etag,
+                RecordType = this.RecordType,
+                IsAutoRegistered = this.IsAutoRegistered
```

Comment:
I think you need to add validation that PATCH/Set record is disallowed for auto-registered record sets. Sync with Jun who added this in service.

Reply:
added check in update.